pkg/controller/serviceaccount/tokens_controller.go

```diff
@@ -431,7 +405,7 @@ func (e *TokensController) ensureReferencedToken(serviceAccount *v1.ServiceAccou
 	}
 	// Manually add the new token to the cache store.
 	// This prevents the service account update (below) triggering another token creation, if the referenced token couldn't be found in the store
-	e.secrets.Add(createdToken)
+	e.secretStore.Add(createdToken)
```

Comment:
I know... but we never resolved https://github.com/kubernetes/kubernetes/pull/39605#discussion_r95491350 so I carried it forward for now.